R/ael04.R

```diff
@@ -0,0 +1,64 @@
+# ael04_main ----
+
+#' @describeIn ael04 Main TLG function
+#'
+#' @inheritParams gen_args
+#' @returns the main function returns an `rlistings` or a `list` object.
+#' @export
+#'
+ael04_main <- modify_default_args(
+  std_listing,
+  dataset = "adsl",
+  key_cols = "ID",
+  disp_cols = c("ASR", "TRTSDTM", "EOSDY", "DTHADY", "DTHCAUS", "ADTHAUT"),
+  split_into_pages_by_var = "ACTARM"
+)
+
+
+#' @describeIn ael04 Preprocessing
+#'
+#' @inheritParams ael04_main
+#' @inheritParams gen_args
+#'
+#' @export
+#'
+ael04_pre <- function(adam_db,
+                      dataset = "adsl",
+                      arm_var = "ACTARM",
```

Comment:
in pre we use `arm_var` however in main we use `split_into_pages_by_var`; what if a user use `run(, arm_var = xxx)` (this will only affect one function)

Reply:
@BFalquet could you please help address this issue, or provide your insights on it?

Reply:
I am a bit split between using `split_into_pages_by_var` as argument name in the preprocessing function (but it doesn't split by page in the preprocessing so it is a bit non sensical) or change the name of the argument in the main function but there is no indication that we will always split the listing by arm variable. I don't really know what is the best solution.

Reply:
fine for me, let's go for that